config/yaml.go

```diff
@@ -41,11 +40,8 @@ type yamlConfigProvider struct {
 }
 
 var (
-	_envSeparator = []byte(":")
-	_emptyString  = []byte(`""`)
-
-	// ${VAR_NAME:default} or ${VAR_NAME}
-	_interpolationRegex = regexp.MustCompile(`\$\{\w+:?[^}]*}`)
+	_envSeparator = ":"
+	_emptyDefault = `""`
```

Comment:
`${PATH}` is currently defined as invalid if the envvar is missing. We shouldn't assume that it's fine that's it's not there and force the service owner to tell us "it's ok if the variable is missing"